modules/concurrency/include/hephaestus/concurrency/io_ring/stoppable_io_ring_operation.h

```diff
@@ -0,0 +1,88 @@
+//=================================================================================================
+// Copyright (C) 2023-2025 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <cerrno>
+#include <system_error>
+
+#include <fmt/format.h>
+#include <liburing.h>
+#include <liburing/io_uring.h>
+#include <stdexec/stop_token.hpp>
+
+#include "hephaestus/concurrency/io_ring/io_ring.h"
+#include "hephaestus/concurrency/io_ring/io_ring_operation_pointer.h"
+#include "hephaestus/utils/exception.h"
+
+namespace heph::concurrency {
+
+template <typename IoRingOperationT>
+struct StoppableIoRingOperation {
```

Comment:
nit. consider moving function implementation into `.cpp`

Reply:
It's a template, but moved the function definitions out of line.